Lib/pathlib/types.py

```diff
@@ -14,10 +14,15 @@
 from glob import _PathGlobber
 from pathlib._os import magic_open, ensure_distinct_paths, ensure_different_files, copyfileobj
 from pathlib import PurePath, Path
-from typing import Optional, Protocol, runtime_checkable
+from typing import (
+    Any, BinaryIO, Callable, Generator, Iterator, Literal, Optional, Protocol, Self, Sequence, TypeVar,
+    runtime_checkable,
+)
 
+_WP = TypeVar("_WP", bound="_WritablePath")
```

Comment:
Would `_CopyTarget` be a better name, now that this is only used from `copy()` and `copy_into()`?

(Not a leading question - I have very little exposure to type vars)

Reply:
I renamed to `_WritablePathT`

Newer syntax would allow us to get rid of the TypeVar import and write:

```
   def copy[T: _WritablePath](self, target: T, **kwargs: Any) -> T:
```

But it'll increase friction with backporting